### PR TITLE
Assignment 6: collecting teammates data through github api and performing search

### PR DESCRIPTION
**Pull request by nitishnivedan**
Assignment 6: collecting teammates data through github api and performing search

### CODE REVIEW

**Comment by nitishnivedan** [comment]
image

**Comment by nitishnivedan** [comment]
image

File: src/Assignment4/NothingFoundComponent/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+import userIcon from "../../logo/userIcon.png";
+import "../CardComponent/styles-card.css";
+
+const CardComponent = ({}) => (
```
Comment (shaksham08) [inline src/Assignment4/NothingFoundComponent/index.js]: you can remove the empty curly braces {}

File: src/Assignment4/CardComponent/index.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const CardComponent = ({
     <div className="information-class">Name : {name}</div>
     <div className="information-class">Place: {location}</div>
     <div className="information-class">Company: {company}</div>
-    <div className="information-class">Designation: {designaton}</div>
+    <div className="information-class">Followers: {designaton}</div>
```
Comment (shaksham08) [inline src/Assignment4/CardComponent/index.js]: here it should be designation or followers? the name is confusing.

**Review by shaksham08 — APPROVED** [review]
Looks good to me , just added some comments

File: src/Assignment4/CardComponent/index.js (hunk 2 above)
Comment (nitishnivedan) [reply src/Assignment4/CardComponent/index.js]: hmm will change the variable name

File: src/apis/userapis.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const getUserInfo = async (userNames) =>
+  await Promise.all(
```
Comment (iswarya15) [inline src/apis/userapis.js:4]: why do we need Promise.all here?

**Review by iswarya15 — COMMENTED** [review]
Looks good to me. Pl check comments

File: src/apis/userapis.js (hunk 3 above)
Comment (nitishnivedan) [reply src/apis/userapis.js:4]: to make concurrent request. so all the apis will be fired parallelly here, overall response time will be lesser.
One more point is concurrent request is done when response of all apis are independent of each other.

**Comment by nikitaj-57** [comment]
Reviewed. Looks good to me.

File: src/Assignment4/CardComponent/styles-card.css [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
-  display: inline-block;
   padding: 10px;
   margin-top: 20px;
   margin-left: 20px;
   margin-right: 20px;
   width: 20%;
-  border: 1px black solid;
+  /* border: 1px black solid; */
+  box-shadow: 0 7px 29px #64646f33;
+  cursor: pointer;
```
Comment (ap221882) [inline src/Assignment4/CardComponent/styles-card.css:9]: You can keep classes names smallcase. Will be clean to write only components as Pascalcase

File: src/apis/userapis.js (hunk 3 above)
Comment (ap221882) [reply src/apis/userapis.js:4]: Proper use of Promise.all. Appreciated!

File: src/Assignment4/Root/index.js [hunk 5]
```diff
@@ -1,28 +1,71 @@
...
-        currentCards.map((temmatesDetails) => (
-          <CardComponent
-            key={temmatesDetails.id}
-            temmatesDetails={temmatesDetails}
-          />
-        ))}
+
+      {currentCards && currentCards.length > 0 ? (
+        <div className="bottomDiv">
+          {currentCards.map((temmatesDetails) => (
```
Comment (ap221882) [inline src/Assignment4/Root/index.js:59]: There is a typo. It is consistent so don't break the code. - teammatesDetails

File: src/apis/userapis.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export const getUserInfo = async (userNames) =>
```
Comment (ap221882) [inline src/apis/userapis.js:3]: Describe your functions. These utility type function's use case aren't easy to find when they are many. Though it's a api only so it is intuitive to find their usecase. But make habit of commenting. I'm also learning that.

File: src/Assignment4/Root/index.js [hunk 7]
```diff
@@ -1,28 +1,71 @@
...
+      "shaksham08",
+      "ap221882",
+      "akshaymarch7",
+    ];
+
+    let users = await getUserInfo(userNames);
+    users =
+      users &&
+      users.map((modifyUserKeys) => ({
+        imageSource: modifyUserKeys.avatar_url,
```
Comment (ap221882) [inline src/Assignment4/Root/index.js]: Here parameter name can be changed. It is confusing to code reviewers. If it was a callback then this name fits I think.

**Review by ap221882 — COMMENTED** [review]
Well done. Some things I also learnt while reviewing. Added some suggestions.

File: src/Assignment4/NothingFoundComponent/index.js (hunk 1 above)
Comment (nitishnivedan) [reply src/Assignment4/NothingFoundComponent/index.js]: removed

File: src/Assignment4/CardComponent/index.js (hunk 2 above)
Comment (nitishnivedan) [reply src/Assignment4/CardComponent/index.js]: done

File: src/Assignment4/CardComponent/styles-card.css (hunk 4 above)
Comment (nitishnivedan) [reply src/Assignment4/CardComponent/styles-card.css:9]: done

File: src/apis/userapis.js (hunk 6 above)
Comment (nitishnivedan) [reply src/apis/userapis.js:3]: done

File: src/Assignment4/Root/index.js (hunk 7 above)
Comment (nitishnivedan) [reply src/Assignment4/Root/index.js]: done

File: src/Assignment4/Root/index.js [hunk 8]
```diff
@@ -1,28 +1,72 @@
...
+      "nitishnivedan",
+      "iswarya15",
+      "smriti-porwal",
+      "shaksham08",
+      "ap221882",
+      "akshaymarch7",
+    ];
+    let users = await Promise.all(
+      userNames.map(async (userNamesInstances) => {
+        const userInformation = await fetch(
```
Comment (ankita1696) [inline src/Assignment4/Root/index.js]: can have the api call outside in a common component.

**Pull request merged**